### PR TITLE
perf: Avoid asyncReadHighestSequenceNr query

### PR DESCRIPTION
**Pull request by patriknw**
* AsyncReplay added in Akka https://github.com/akka/akka/pull/32434

### CODE REVIEW

File: samples/guide/shopping-cart-java/shopping-analytics-service/pom.xml [hunk 1]
```diff
@@ -43,106 +53,116 @@
...
-        <dependencies>
-            <dependency>
-                <groupId>com.lightbend.akka</groupId>
-                <artifactId>akka-dependencies_${scala.binary.version}</artifactId>
-                <version>24.05.0</version>
-                <type>pom</type>
-                <scope>import</scope>
-            </dependency>
-        </dependencies>
-    </dependencyManagement>
```
Comment (patriknw) [inline samples/guide/shopping-cart-java/shopping-analytics-service/pom.xml:56]: transitive snapshot dependencies doesn't work with dependencyManagement

File: core/src/main/scala/akka/persistence/dynamodb/journal/DynamoDBJournal.scala [hunk 2]
```diff
@@ -181,34 +188,68 @@ private[dynamodb] final class DynamoDBJournal(config: Config, cfgPath: String) e
...
+          .internalCurrentEventsByPersistenceId(persistenceId, fromSequenceNr, toSequenceNr, includeDeleted = true)
+          .runWith(Sink.foreach { item =>
+            seqNr.set(item.seqNr)
+            // payload is empty for deleted item
+            if (item.payload.isDefined) {
+              val repr = deserializeItem(serialization, item)
+              recoveryCallback(repr)
+            }
+          })
+          .map(_ => seqNr.get)(ExecutionContexts.parasitic)
```
Comment (pvlugter) [inline core/src/main/scala/akka/persistence/dynamodb/journal/DynamoDBJournal.scala]: Atomic long preferable over doing something like a fold?

```scala
.runWith(Sink.fold(0L) { (_, item) =>
  // ...
  item.seqNr
})

**Review by pvlugter — COMMENTED** [review]
Looks good

File: core/src/main/scala/akka/persistence/dynamodb/journal/DynamoDBJournal.scala (hunk 2 above)
Comment (patriknw) [reply core/src/main/scala/akka/persistence/dynamodb/journal/DynamoDBJournal.scala]: that's better, thanks

**Comment by patriknw** [comment]
Updated with Akka 2.9.4

**Review by pvlugter — APPROVED** [review]
LGTM

**Pull request merged**